estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -510,55 +525,140 @@ def open_binding(
+                    subtask_id=subtask_id,
+                ):
+                    assert state.inc and isinstance(state.inc, dict)
+                    inc_state = state.inc.get(subtask_id)
+                    assert inc_state
+
+                    await _binding_incremental_task(
+                        binding,
+                        binding_index,
+                        fetch_fn,
+                        inc_state,
+                        subtask_task,
+                    )
 
-        async def closure(task: Task):
-            assert state.inc
-            await _binding_incremental_task(
-                binding, binding_index, fetch_changes, state.inc, task,
-            )
+                task.spawn_child(
+                    f"{prefix}.incremental.{subtask_id}", incremental_subtask
+                )
+        else:
+
+            async def closure(task: Task):
+                assert state.inc and not isinstance(state.inc, dict)
+                await _binding_incremental_task(
+                    binding,
+                    binding_index,
+                    fetch_changes,
+                    state.inc,
+                    task,
+                )
 
-        task.spawn_child(f"{prefix}.incremental", closure)
+            task.spawn_child(f"{prefix}.incremental", closure)
 
     if fetch_page and state.backfill:
+        if isinstance(fetch_page, dict):
+            for subtask_id, subtask_fetch_page in fetch_page.items():
+
+                async def backfill_subtask(
+                    subtask_task: Task,
+                    fetch_fn=subtask_fetch_page,
+                    subtask_id=subtask_id,
+                ):
+                    assert state.backfill and isinstance(state.backfill, dict)
+                    backfill_state = state.backfill.get(subtask_id)
+                    assert backfill_state
+
+                    await _binding_backfill_task(
+                        binding,
+                        binding_index,
+                        fetch_fn,
+                        backfill_state,
+                        subtask_task,
+                    )
 
-        async def closure(task: Task):
-            assert state.backfill
-            await _binding_backfill_task(
-                binding, binding_index, fetch_page, state.backfill, task,
-            )
+                task.spawn_child(f"{prefix}.backfill.{subtask_id}", backfill_subtask)
+        else:
+
+            async def closure(task: Task):
+                assert state.backfill and not isinstance(state.backfill, dict)
+                await _binding_backfill_task(
+                    binding,
+                    binding_index,
+                    fetch_page,
+                    state.backfill,
+                    task,
+                )
 
-        task.spawn_child(f"{prefix}.backfill", closure)
+            task.spawn_child(f"{prefix}.backfill", closure)
 
     if fetch_snapshot:
+        if isinstance(fetch_snapshot, dict):
```

Comment:
As noted above, I don't think we should attempt to support multiple cursors for snapshot tasks right now.

Reply:
Updates to remove this.